src/CAN/can_mapping.c

```diff
@@ -32,6 +32,8 @@ float canmapping_extract_value(uint64_t raw_data, const CANMapping *mapping)
         }
         uint32_t bitmask = (1UL << length) - 1;
         uint32_t raw_value = (raw_data >> offset) & bitmask;
+
+        /* normalize endian */
```

Comment:
Again this always makes the assumption you are on a little endian system.  This is potentially dangerous.  Do you at least have a check to ensure that we are a little endian system and blow up if we are not?  Would be wise.